crates/krun/src/bin/krun.rs

```diff
@@ -62,17 +62,59 @@ fn main() -> Result<()> {
     };
 
     {
+        let cpus = if let Some(cpus) = options.cpus {
+            cpus
+        } else {
+            let mut perf_max_freq = None;
+            let mut perf_core_count = 0u16;
+            let dir_iter = fs::read_dir("/sys/devices/system/cpu")
+                .context("Failed to read directory `/sys/devices/system/cpu`")?;
+
+            for entry in dir_iter {
+                let entry =
+                    entry.context("Failed to read directory entry in `/sys/devices/system/cpu`")?;
+                let file_name = entry
+                    .file_name()
+                    .into_string()
+                    .expect("file_name should not contain invalid UTF-8");
+                let Some(_core_num) = file_name
+                    .strip_prefix("cpu")
+                    .and_then(|suffix| suffix.parse::<u16>().ok())
+                else {
+                    // e.g. `cpufreq`, `cpuidle`
+                    continue;
+                };
+                let path = entry.path().join("cpufreq/cpuinfo_max_freq");
```

Comment:
Do we need to handle the case of `cpuinfo` being unavailable?

Reply:
Actually, I was mistaken, this is from `cpufreq`. Anyway, this should be okay now that we have `get_fallback_cores`.